### ISSUE TITLE
Bug in Dataset method save_dataset()

### ISSUE THREAD

**Issue by martinholmer** [PolicyEngine/policyengine-core#104]
TWO THINGS: what seems to be a bug and a feature request:

BUG:
In the Dataset class, `exists` is a property not a method.
So, should the error-causing line be:
```
with h5py.File(file, "a" if file.exists else "w") as f:
```
ERROR MESSAGE:
```
vdset.save_dataset(vdset)
  File "//Users/mrh/anaconda3/envs/policyengine-us/lib/python3.9/site-packages/policyengine_core/data/dataset.py", line 172, in save_dataset
    with h5py.File(file, "a" if file.exists() else "w") as f:
AttributeError: 'str' object has no attribute 'exists'
```

FEATURE REQUEST:
Also, could you add an optional argument to the `save_dataset` method that allows use of a file path that differs from `self.file_path`?

**Comment by martinholmer** [comment]
@nikhilwoodruff,  I can't assign #104 to you or label it a bug.
Can you fix this on Friday?

**Comment by nikhilwoodruff** [comment]
Hi @martinholmer - just taking a look now. Here's what I understand:

* The `file` object in your examples is the `file_path` attribute of `Dataset`, which is actually a `Path` object. `Path` does have the `exists()` function, so I think the Core code is correct.  But just to be safe, I've added an extra check in `save_dataset` that casts `file_path` to a `Path` in case it's a `str`. 
* I've also added the optional argument for a custom filepath in that method.

These changes are in #105.
